### PR TITLE
create likes page & api

### PR DESCRIPTION
**Pull request by git-0r**
### Likes page and api calls

video-lib-likes

### Files to review:
 - context/likesContext
 - pages/likes
 - pages/video
 - api-calls

### CODE REVIEW

File: src/components/LoginForm.js [hunk 1]
```diff
@@ -12,6 +13,7 @@ const LoginForm = () => {
     const navigate = useNavigate();
     const { notificationHandler } = useNotification();
     const { updateWatchLater } = useWatchLater();
+    // const { updateLikes} = useLikes();
```
Comment (nikhiljugale007) [inline src/components/LoginForm.js]: suggestion: remove commented code if not required

File: src/components/VideoListingCard.js [hunk 2]
```diff
@@ -34,6 +36,17 @@ const VideoListingCard = ({ video }) => {
         }
     }
 
+    const updateHistory = async () => {
+        const updatedHistory = await removeFromHistory(video._id, user);
+        setHistory(updatedHistory);
+        notificationHandler("Removed from history");
+        try {
+
```
Comment (nikhiljugale007) [inline src/components/VideoListingCard.js:40]: try block is empty. it is intended or you planning to do write in future.

**Review by nikhiljugale007 — APPROVED** [review]
Did not find any major issues. Good to merge.

File: src/App.js [hunk 3]
```diff
@@ -18,18 +21,23 @@ const App = () => {
...
-                            <Route path="/watchlater/:id" element={<WatchLater />} />
-                            <Route path="/video/:id" element={<Video />} />
-                            <Route path="/history/:id" element={<History />} />
-                        </Routes>
-                    </BrowserRouter>
-                </WatchLaterContextProvider>
+                <WatchLaterProvider>
+                    <LikesProvider>
+                        <HistoryProvider>
+                            <BrowserRouter>
```
Comment (yaswanthmyneni) [inline src/App.js:27]: suggestion: give providers in the index.js file instead of here.

File: src/App.js [hunk 4]
```diff
@@ -18,18 +21,23 @@ const App = () => {
...
+                                <Routes>
+                                    <Route path="/" element={<Home />} />
+                                    <Route path="/auth/login" element={<Login />} />
+                                    <Route path="/auth/register" element={<Signup />} />
+                                    <Route path="/watchlater/:id" element={<WatchLater />} />
+                                    <Route path="/video/:id" element={<Video />} />
+                                    <Route path="/history/:id" element={<History />} />
+                                    <Route path="/likes/:id" element={<Likes />} />
```
Comment (yaswanthmyneni) [inline src/App.js:35]: suggestion: create separate component for routes.

File: src/pages/Likes.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
```
Comment (yaswanthmyneni) [inline src/pages/Likes.js:1]: changes: I cannot see removing liked video cards from the likes page. I think it should be present.

**Review by yaswanthmyneni — COMMENTED** [review]
you can merge it after changes.

File: src/components/LoginForm.js (hunk 1 above)
Comment (git-0r) [reply src/components/LoginForm.js]: removed.

File: src/components/VideoListingCard.js (hunk 2 above)
Comment (git-0r) [reply src/components/VideoListingCard.js:40]: fixed the empty try block.

File: src/pages/Likes.js (hunk 5 above)
Comment (git-0r) [reply src/pages/Likes.js:1]: as per user stories in lms, likes page or option to remove a liked video from likes page isn't required. So, i'll skip it for now.

**Pull request merged**